### DISCUSSION TITLE
URGENT: MY PC STOPPED WORKING AFTER THE SETUP

### DISCUSSION THREAD

**Discussion by emanuelecc** [valinet/ExplorerPatcher#1221] [Q&A]
https://user-images.githubusercontent.com/129448447/228966382-a2d67548-6000-40e8-afcf-2bbbd8fbc4e7.mp4

Hi, I installed the ep_setup.exe file on my Asus ZenBook 14 UM5401QA with windows 11 and the result is what you see in the attached video: the computer is unusable, it keeps flashing the screen and I can't even open the start menu to uninstall it, I can't open the command prompt nor the settings to format the pc. Please help me because the computer arrived today and I can't use it to work!! 
How can I uninstall your program? The only thing that works is "Ctrl Alt Canc" but I can't find ep_setup among the working app section. 

Thank you in advance
Emanuele

**Comment by emanuelecc** [comment]
These are the ongoing process. I can't even access to prompt, please I'm desperate!! 

https://user-images.githubusercontent.com/129448447/228968185-5466b5de-4328-4b60-adf1-f9a16b23adb1.mp4

**Comment by Stojakovic83** [comment]
That problem is caused by a Windows update. You need to restart and press F8, enter troubleshoot, and uninstall the latest quality update.

**Reply by emanuelecc** [reply]
When I press F8 nothing happens. How can I enter troubleshoot? It's and Asus ZenBook UM5401QA with Windows 11. Anyway the problem started after I installed the ep_setup.exe, not after un update

**Reply by Stojakovic83** [reply]
Try to do it this way, turn on the pc and in the meantime press the reset button (do this operation 3 times) and then you will enter the troubleshooter

**Reply by Intenzi** [reply]
I have faced the same issue, I have uninstalled the latest quality update and it fixed the flashing and let my computer run.
However it seems to have caused my taskbar to have various issues which may or may not be due to the windows update?
However nothing seems to fix them so I have considered them to be taskbar problems.
1. I cannot peek open app's previews from my taskbar, it simply shows their current title. (have tried regedit, etc, etc.)
2. I have dual monitor setup and my display is now scuffed wherein my first display (a laptop) takes 50% of the left screen size starting from the bottom of the second monitor but now moving mouse pointer anywhere above it in my second monitor also jumps the mouse pointer to the first monitor.

A minor fault that might assist? I have noticed is that sometimes due to certain actions the icons in the taskbar disappear, which I assume would normally be refresh, but now they only get refreshed icon by icon by dragging my mouse across the entire taskbar so that they load up the icons.

**Reply by ghost** [reply]
Hold down the Left Shift key while you tell the PC to restart. It should send you to a blue troubleshooting screen, select the option for Safe Mode, then allow the PC to restart. Once in safe mode, uninstall EP or roll back the Windows Update using System Restore.

**Reply by emanuelecc** [reply]
I've already tried it last week, but it didn't work, I couldn't uninstall EP because the flashing was going on in Safe Mode too. And I couldn't event uninstall the last update. I had to go to an Asus store and the technician tried to do anything for two hours, but nothing has worked. 

He had to reinstall Windows, loosing all my data. I regret to have installed this program, I lost everything.

**Comment by pyrates999** [comment]
Sorry to hear this happened to you.  As someone who does do troubleshooting, reinstalling windows by wiping the drive is very drastic.  Windows does have a reinstall option that doesn't wipe your data.  Check the directory C:\Users\ and see if your old username is there.  If it is, and you open it, it'll ask you for permission to allow you access.  This may take a while, so just let it sit and do it's thing.  When it's done, it will open up the directory and you'll see what's inside.

I found this on how to uninstall explorer patcher:

https://github.com/valinet/ExplorerPatcher/wiki#uninstalling

But it uses the automatic method.

It's been discussed how to disable explorer patcher from starting up.  Could someone include the steps needed to do that under a Manual Uninstall section?  I think linking to that would be very helpful for someone in the future that is stuck like this.  It could include how to disable explorer patcher and would work even when you boot from windows 10/11 install media using the command line.  It also would include what needs to be removed from the registry to also do a manual uninstall.

**Comment by GerardSmit** [comment]
If anyone else has this problem, you can uninstall Explorer Patcher by the following method (even with the flashing screen):

1. Open "Run" with [Windows Key] + [R]
2. If you do not have the installation on your computer:
  a. Enter "msedge" or "iexplore" and click on "OK"
  b. Go to https://github.com/valinet/ExplorerPatcher/releases
  c. Download the latest release
  d. Re-open "Run" with [Windows Key] + [R]
3. Enter "cmd" and press enter
4. Go to the directory where the setup file is located with cd.  
    For example, if it's in your downloads: `cd Downloads` and press enter.
6. Run the following command: `ep_setup.exe /uninstall` and press "Yes"

**Reply by pyrates999** [reply]
That's what the link says to do.  I'm referring to a manual uninstall.  What files do I need to remove from the file system that explorer patcher installs?

**Reply by GerardSmit** [reply]
Seems like you can also just remove `dxgi.dll` in C:\Windows
>  ExplorerPatcher had an update to address that. To stop the crash loop, delete dxgi.dll in C:\Windows.
> https://github.com/valinet/ExplorerPatcher/issues/1148#issuecomment-1476323176

**Reply by topia** [reply]
> Seems like you can also just remove `dxgi.dll` in C:\Windows
> >  ExplorerPatcher had an update to address that. To stop the crash loop, delete dxgi.dll in C:\Windows.
> > https://github.com/valinet/ExplorerPatcher/issues/1148#issuecomment-1476323176
> 

Those steps are for "stopping explorer crash loop" only. you should run uninstaller after stopping.

**Reply by pyrates999** [reply]
I would not go around deleting files that came with windows such as `dxgi.dll` in C:\Windows.  I'd rather the bug was fixed first.  At the very least, just rename the file instead of deleting it, so you can recover it later.

**Reply by topia** [reply]
> I would not go around deleting files that came with windows such as `dxgi.dll` in C:\Windows.  I'd rather the bug was fixed first.  At the very least, just rename the file instead of deleting it, so you can recover it later.

It's coming from ExplorerPatcher, not windows. but the same named file in system32 is coming from windows.

**Reply by ankur-gadgilwar109** [reply]
This worked. Thank you! However, it's difficult typing in the terminal.

**Comment by kleenkanteen** [comment]
Btw this same issue happened with me. I reinstalled windows 11 then it worked. Just found out this is causing the issue. I didn't have to wipe my laptop though, should've uninstalled just explorerpatch.

**Comment by morphiya** [comment]
Finally! Solved it 🤬 I couldn't do anything except running task manager (Crtl+Alt+Del), Win+R didn't work. So I ran a new process in task manager - cmd and finally got an access to the command line. Moved to folder with ep_setup.exe and ran `ep_setup.exe /uninstall `. It was an hour of hell.

**Comment by JaielZeus** [comment]
This same happened to me with the march update already. I had to restore my pc to the point before the update was applied, which is automatically set as a restore point by windows good god. Then today I did a manual restore point because I saw windows wanted to update again after restart, but this was an error because I fucked myself over. The automatic restore point set today was 4 hours earlier but I did a lot of work that I didnt sync with github yet so I couldn't afford to go back 4 hours earlier and the restore point I did manually did not work because as soon as I restore to this point and start windows it would update.

I went the route of uninstalling the latest quality update which worked out (open ctrl+shift+del selection menu and shift click on restart button and select the option from the debug menu). I am sure @morphiya 's solution would have worked also by running cmd from taskmanager and manually uninstall Explorer patcher.

Anyways I see 3 possible solutions:
- if you can afford go back to the restore point before the update hit by shift+clicking on restart, restore to a previous point and then update your Eplorer Patcher.
- Uninstall the latest quality updates by shift-clicking on restart and then update your explorer patcher
- Or start cmd.exe from task manager and manually uninstall explorer patcher (and reinstall new updated one)

This is kinda bad imo that Explorer Patcher fucks this new update up on one hand but also from windows delivering an update which breaks things also. I am glad to have found a solution after a month finally

**Comment by RDMaven** [comment]
Had the same thing and i can assure you I won't be using this """patcher""" anymore.
I didn't have ep_setup.exe installed so I used the task manager to create a cmd session and got the latest version with curl then for some reason the executable couldn't be downloaded from the terminal so I launched an browser session and downloaded the script. After that, the uninstall command got rid of this extension for good.
Thank you to the people who shared their methods to solve this issue !

**Reply by JaielZeus** [reply]
I hear your frustration but don't be too mad about explorer patcher. Wiindows 11 update came with a lot of bugs on their own.

**Comment by anton-epishev** [comment]
Is this issue fixed in newer versions? Could not find this is released. I can't uninstall security update, as this is managed by org. But I also can't live with that grouped apps icons...

**Reply by pyrates999** [reply]
I wouldn't install explorer patcher on a work computer.  Leave it for your home computer, unless you know what you're doing and know how to uninstall it if necessary.

**Reply by JaielZeus** [reply]
It is fixed. You HAVE to uninstall the old version first though and then fresh install. I wouldn't go too bad on the explorer patcher for this issue though as the fix is easy. The next bigger update of windows will include the functionality to ungroup windows in the taskbar.

**Reply by anton-epishev** [reply]
Thanks, it is fixed indeed.

**Comment by pyrates999** [comment]
I believe this is fixed.  If explorer patcher causes explorer to crash 3 times, it will show an error message indicating it is not loaded, and then you can uninstall it.

**Reply by paulius-valiunas** [reply]
this fix doesn't work, I just entered a crash loop, it showed me the error message you are talking about but it disappeared after a few seconds and explorer restarted again. So in effect I just had an error message appear every 3rd restart but that didn't change anything.

**Reply by pyrates999** [reply]
After the error message appears, you are supposed to then uninstall explorer patcher.

**Reply by paulius-valiunas** [reply]
but you can't because explorer still keeps crashing. The error message does not help in any way other than tell you to uninstall. I did eventually manage to launch control panel from task manager (win+r and other shortcuts didn't work), but someone less tech-savvy than me would find it much more problematic.

**Reply by paulius-valiunas** [reply]
my point is, the error message _says_ that it will stop crashing now so you can uninstall it safely, but it doesn't stop crashing and your PC keeps lagging and being generally unresponsive. So, it's not at helpful as it should be.

**Reply by pyrates999** [reply]
Do you try to close it after it shows and it keeps showing?  Or when it shows, do you then try to uninstall explorer patcher?

The latest release 22621.1555.55.1 is supposed to only show it once and then it stops until you reboot.  Do you have an earlier version installed when this was happening?

**Reply by paulius-valiunas** [reply]
> Do you try to close it after it shows and it keeps showing? Or when it shows, do you then try to uninstall explorer patcher?
> 
> The latest release 22621.1555.55.1 is supposed to only show it once and then it stops until you reboot. Do you have an earlier version installed when this was happening?

it doesn't matter what I do, it disappears after about 10 seconds. It doesn't react to me clicking anything, not sure if it's a bug in explorer patcher or just general unresponsiveness of the computer. I don't remember which version I had installed, I didn't have time to collect troubleshooting information because I was busy trying to get explorer.exe working as soon as possible.

**Reply by atqamz** [reply]
after update windows, try to reinstall ExplorerPatcher, if you can't access it via taskbar or else,

try to open Task Manager by pressing ctrl + shift + esc, then run new task 'cmd', then navigate to where the ep_setup.exe placed on your computer, then type 'ep_setup.exe /uninstall', wait for a while and your taskbar will running normally

**Reply by paulius-valiunas** [reply]
> after update windows, try to reinstall ExplorerPatcher, if you can't access it via taskbar or else,
> 
> try to open Task Manager by pressing ctrl + shift + esc, then run new task 'cmd', then navigate to where the ep_setup.exe placed on your computer, then type 'ep_setup.exe /uninstall', wait for a while and your taskbar will running normally

Were you replying to me? As I said:
> I did eventually manage to launch control panel from task manager (win+r and other shortcuts didn't work), but someone less tech-savvy than me would find it much more problematic.

**Comment by octipus** [comment]
Jesus fucking christ? Broke my keyboard and almost smashed the display. This "tool" should be wiped out from existence. Worst debugging ever

**Comment by nachito9380** [comment]
I've gotten the flashing screen problem too. I've managed to open troubleshooting and I'm at the command prompt. I don't know how to uninstall explorer patcher. Can someone help?

**Reply by vippium** [reply]
Steps to uninstall from cmd :
1. open cmd
2. type notepad
3. in notepad, go to File --> Open
4. navigate to the download folder where ep in downloaded
5. Select allfiles in file type from .txt
6. copy the path
7. switch to cmd and type `cd ---path---`   (simply r-click in console after typing cd)
8. then type ep_setup.exe /uninstall

It will unistall EP from cmd

**Reply by nachito9380** [reply]
Like this?
![IMG_20230729_073809517](https://github.com/valinet/ExplorerPatcher/assets/140870873/5972a42b-db03-46f1-b52d-a400aa123b2f)

**Comment by nachito9380** [comment]
> Steps to uninstall from cmd :
> 
> 1. open cmd
> 2. type notepad
> 3. in notepad, go to File --> Open
> 4. navigate to the download folder where ep in downloaded
> 5. Select allfiles in file type from .txt
> 6. copy the path
> 7. switch to cmd and type `cd ---path---`   (simply r-click in console after typing cd)
> 8. then type ep_setup.exe /uninstall
> 
> It will unistall EP from cmd

Ignore my previous reply. I tried everything and nothing worked. Eventually, I gave up and ended up resetting my PC with a fresh install of windows 11. That seems to have fixed the problem. At least all my files are safe and sound. Thanks for helping anyways.

**Comment by Twisted-Code** [comment]
I had these exact symptoms (Explorer crash-looping as seen in OP's video and being unable access much of the Windows shell faculties as a result) a little less than a year ago. I can't speak to my problem-solving process at the time, since I'd been through a lot of shit already with the laptop in question (never buy a Dell, Alienware or otherwise) and was looking for an excuse to [defenestrate](https://www.wordnik.com/words/defenestration) it. However, I do think it's noteworthy that at one point in problem solving/excuse seeking process, I tried safe mode and found that this flashing behavior persisted. Does EP still run in safe mode, and if so, why?

**Reply by pyrates999** [reply]
This sounds like your windows install was corrupt.

**Reply by paulius-valiunas** [reply]
ExplorerPatcher is a very hacky solution, I think it replaces some DLLs so you can't disable it without replacing them again with the original DLLs. Safe mode doesn't do that, it just disables 3rd party startup scripts/executables that are registered using conventional methods. You can uninstall ExplorerPatcher without entering safe mode: press ctrl+shift+esc to bring up task manager, click the "run new task" button, enter "control" in the box (without quotes) and click OK. Then just uninstall it as usual.

**Reply by pyrates999** [reply]
If that's the case, you will have to repair your windows install after uninstalling EP and you don't intend to install EP again.

**Reply by paulius-valiunas** [reply]
no you won't, the (un)installer puts everything back where it belongs

**Reply by pyrates999** [reply]
Well alrighty then.

**Reply by Twisted-Code** [reply]
I'm trying to think whether at any point I managed to get the computer to a command line. I know I was able to get Windows recovery options to come up, does that have a "open your REPL shell so I can fix you, stupid" button?

**Comment by brtbendele** [comment]
I'm really struggling with this and gotten to the point where I just needs to uninstall this from my work computer... but I cannot. I just did the latest windows update and this is when it happened.  the ep_setup.exe / uninstall command has never gotten me anywhere. I get the same response every time when I run it in windows. 
![image](https://github.com/valinet/ExplorerPatcher/assets/56659041/f6841e7a-6f2a-4a9a-a056-38e6698804bb)
And in Powershell (admin) or Terminal 
![image](https://github.com/valinet/ExplorerPatcher/assets/56659041/0f60715b-7c44-4ae3-b47d-05dee1b98ebe)

**Reply by pyrates999** [reply]
You may need to repair your windows install.  Please run the following in an elevated command prompt or elevated powershell:

```
chkdsk C:
```

If it tells you there are errors, then run this:

```
chkdsk C: /F
```

Reboot to run it on boot.  Keep running that command until it reports no errors.  Then run this to repair windows:

```
sfc /scannow
dism /online /cleanup-image /scanhealth
dism /online /cleanup-image /restorehealth
dism /online /cleanup-image /checkhealth
dism /online /cleanup-image /analyzecomponentstore
dism /online /cleanup-image /startcomponentcleanup
```

Then reboot your computer.

**Reply by pyrates999** [reply]
and see here for ways to uninstall after: https://github.com/valinet/ExplorerPatcher/wiki/Installer-How-To#how-to-uninstall

**Reply by pyrates999** [reply]
I would also not be installing EP on any work computer since you could lose administrator privileges at anytime due to IT.  Without that, you can't uninstall, install, or update EP.

**Reply by paulius-valiunas** [reply]
> I would also not be installing EP on any work computer since you could lose administrator privileges at anytime due to IT. Without that, you can't uninstall, install, or update EP.

Please elaborate "due to IT". If I ever lose admin privileges on my work machine, I'll go rip a new one to the IT admin that did it, saying it blocked me from doing my job. Easy solution.

**Reply by pyrates999** [reply]
You should not need admin privileges to do your work unless your work involves installing and uninstalling software.  It's also more secure not giving everyone admin privileges, only to those who need it.  And unless you work in IT, you probably don't need it.

**Reply by paulius-valiunas** [reply]
I'm a developer, so yes, my job requires me to install/uninstall software. And I imagine most people installing EP are somewhat IT-oriented and have jobs in related fields. Of course the most secure solution is to not give anyone a computer at all.

**Reply by pyrates999** [reply]
Lets check if the file ep_setup.exe has completely been downloaded.  Right click on it and select properties.  Please show screenshots of the general, compatibility, and details tab.
